ios/testmanagerd/testlistener.go

```diff
@@ -233,13 +252,25 @@ func (t *TestListener) Done() <-chan struct{} {
 	return t.executionFinished
 }
 
-func (ts *TestSuite) findTestCase(className string, methodName string) *TestCase {
-	for i, _ := range ts.TestCases {
-		tc := &ts.TestCases[i]
-		if tc.ClassName == className && tc.MethodName == methodName {
-			return tc
+func (t *TestListener) findTestCase(className string, methodName string) *TestCase {
+	for i, _ := range t.TestSuites {
+		ts := &t.TestSuites[i]
+		for j, _ := range ts.TestCases {
+			tc := &ts.TestCases[j]
+			if tc.ClassName == className && tc.MethodName == methodName {
+				return tc
+			}
 		}
 	}
 
 	return nil
 }
+
+func (t *TestListener) findTestSuite(className string) *TestSuite {
```

Comment:
Oops, branch had missing files from my local. Since tests didn't catch this, I also added additional assertions to make sure this cannot happen again. Thank you for pointing it out.

Pushed the fix.